session.go

```diff
@@ -223,10 +222,18 @@ func (s *session) doCommit() error {
 }
 
 func (s *session) doCommitWithRetry() error {
+	var txnSize int
+	if s.txn != nil && s.txn.Valid() {
+		txnSize = s.txn.Size()
+	}
 	err := s.doCommit()
 	if err != nil {
 		if s.isRetryableError(err) {
-			err = s.Retry()
+			// Transactions will retry 2 ~ 10 times.
+			// We make larger transactions retry less times to prevent cluster resource outage.
+			txnSizeRate := float64(txnSize) / float64(kv.BufferSizeLimit)
```

Comment:
@shenli 
Then it will not do any retry.
`-1` doesn't mean unlimited here anymore.